src/app/CommandSender.h

```diff
@@ -114,6 +114,33 @@ class CommandSender final : public Messaging::ExchangeDelegate
         virtual void OnDone(CommandSender * apCommandSender) = 0;
     };
 
+    // While today there is only one item in this struct, this was added to make providing future elements of
+    // InvokeRequests easier.
+    struct AdditionalInvokeRequestElements
+    {
+        Optional<uint16_t> commandRef;
+    };
+
+    // While today there is only one item in this struct, this was added to make providing future elements of
+    // InvokeResponse easier.
+    struct AdditionalInvokeResponseElements
+    {
+        Optional<uint16_t> commandRef;
+    };
+
+    /**
+     * Called from Callbacks to get additional elements in InvokeResponseMessage.
+     *
+     * For legacy reasons we cannot change arguments passed through `OnResponse`, but we wanted to provide a
+     * way for arguments to be queried going forward from within the `OnResponse` callback.
+     *
+     * @param [out] aInvokeReponseElements additional InvokeResponse elements.
+     *
+     * @return CHIP_ERROR_INCORRECT_STATE
+     *                      If CommandSender is not in ResponseReceived state.
+     */
+    CHIP_ERROR GetAdditionalInvokeResponseElements(AdditionalInvokeResponseElements & aInvokeReponseElements);
+
```

Comment:
I am not entirely sure how true this assumption is. If this is not correct I will remove this and add `AdditionalInvokeResponseElements` to the callback parameters. Please just let me know